crates/consensus/beacon/src/engine/invalid_headers.rs

```diff
@@ -79,11 +80,11 @@ impl<H: Debug> InvalidHeaderCache<H> {
     }
 }
 
-struct HeaderEntry<H> {
+struct HeaderEntry {
     /// Keeps track how many times this header has been hit.
     hit_count: u8,
-    /// The actually header entry
-    header: Arc<H>,
+    /// The actual header entry
+    header: Arc<BlockWithParent>,
```

Comment:
I think `Arc` here is unnecessary but can keep as is for now until we have `Copy` https://github.com/alloy-rs/alloy/pull/1776

Reply:
just changed it to clone, we can remove the clone once we have the `Copy` impl